File: enterprise/health-checks/health-check.js
#!/usr/bin/env node

/**
 * TuskLang Health Check System - JavaScript/Node.js Implementation
 * Comprehensive health monitoring for all TuskLang services
 */

const https = require('https');
const http = require('http');
const { promisify } = require('util');
const fs = require('fs').promises;
const os = require('os');

// Health status enumeration
const HealthStatus = {
    HEALTHY: 'healthy',
    DEGRADED: 'degraded',
    UNHEALTHY: 'unhealthy',
    UNKNOWN: 'unknown'
};

/**
 * Health check result
 */
class HealthCheck {
    constructor(name, status, message, details = null) {
        this.name = name;
        this.status = status;
        this.message = message;
        this.details = details;
        this.timestamp = new Date().toISOString();
    }
}

/**
 * Health report
 */
class HealthReport {
    constructor() {
        this.timestamp = new Date().toISOString();
        this.overall_status = HealthStatus.HEALTHY;
        this.checks = [];
        this.summary = {
            total: 0,
            healthy: 0,
            degraded: 0,
            unhealthy: 0,
            unknown: 0
        };
    }

    addCheck(check) {
        this.checks.push(check);
        this.summary.total++;
        
        switch (check.status) {
            case HealthStatus.HEALTHY:
                this.summary.healthy++;
                break;
            case HealthStatus.DEGRADED:
                this.summary.degraded++;
                if (this.overall_status === HealthStatus.HEALTHY) {
                    this.overall_status = HealthStatus.DEGRADED;
                }
                break;
            case HealthStatus.UNHEALTHY:
            case HealthStatus.UNKNOWN:
                this.summary.unhealthy++;
                this.overall_status = HealthStatus.UNHEALTHY;
                break;
        }
    }
}

/**
 * TuskLang Health Checker
 */
class TuskLangHealthChecker {
    constructor(config) {
        this.config = config;
    }

    /**
     * Check system resources
     */
    async checkSystemResources() {
        try {
            const totalMem = os.totalmem();
            const freeMem = os.freemem();
            const memoryUsage = ((totalMem - freeMem) / totalMem) * 100;
            
            const cpus = os.cpus();
            const cpuUsage = cpus.reduce((acc, cpu) => {
                const total = Object.values(cpu.times).reduce((a, b) => a + b);
                const idle = cpu.times.idle;
                return acc + ((total - idle) / total) * 100;
            }, 0) / cpus.length;

            // Get disk usage (simplified - in production use a proper disk usage library)
            const diskUsage = 75; // Simulated for now

            const details = {
                cpu_percent: cpuUsage.toFixed(2),
                memory_percent: memoryUsage.toFixed(2),
                disk_percent: diskUsage,
                memory_available: freeMem,
                cpu_count: cpus.length
            };

            let status = HealthStatus.HEALTHY;
            let message = 'System resources are normal';

            if (cpuUsage > 80 || memoryUsage > 85 || diskUsage > 90) {
                status = HealthStatus.DEGRADED;
                message = `High resource usage - CPU: ${cpuUsage.toFixed(1)}%, Memory: ${memoryUsage.toFixed(1)}%, Disk: ${diskUsage}%`;
            }

            return new HealthCheck('system_resources', status, message, details);
        } catch (error) {
            return new HealthCheck('system_resources', HealthStatus.UNHEALTHY, `System resource check failed: ${error.message}`);
        }
    }

    /**
     * Check database connectivity
     */
    async checkDatabase() {
        try {
            const dbConfig = this.config.database;
            if (!dbConfig) {
                return new HealthCheck('database', HealthStatus.UNHEALTHY, 'Database configuration not found');
            }

            // In a real implementation, you would use a database client
            // For now, we'll simulate the check
            const isConnected = true; // Simulated
            const activeConnections = 25; // Simulated
            const dbSize = 1024 * 1024 * 100; // 100MB simulated

            const details = {
                active_connections: activeConnections,
                database_size_bytes: dbSize,
                database_size_mb: Math.round(dbSize / 1024 / 1024)
            };

            let status = HealthStatus.HEALTHY;
            let message = 'Database is healthy';

            if (!isConnected) {
                status = HealthStatus.UNHEALTHY;
                message = 'Database connection failed';
            } else if (activeConnections > 100) {
                status = HealthStatus.DEGRADED;
                message = `High number of active connections: ${activeConnections}`;
            }

            return new HealthCheck('database', status, message, details);
        } catch (error) {
            return new HealthCheck('database', HealthStatus.UNHEALTHY, `Database check failed: ${error.message}`);
        }
    }

    /**
     * Check Redis connectivity
     */
    async checkRedis() {
        try {
            const redisConfig = this.config.redis;
            if (!redisConfig) {
                return new HealthCheck('redis', HealthStatus.UNHEALTHY, 'Redis configuration not found');
            }

            // In a real implementation, you would use a Redis client
            // For now, we'll simulate the check
            const isConnected = true; // Simulated
            const memoryUsagePercent = 45; // Simulated
            const connectedClients = 15; // Simulated

            const details = {
                memory_usage_percent: memoryUsagePercent,
                connected_clients: connectedClients,
                used_memory_mb: 512,
                max_memory_mb: 1024
            };

            let status = HealthStatus.HEALTHY;
            let message = 'Redis is healthy';

            if (!isConnected) {
                status = HealthStatus.UNHEALTHY;
                message = 'Redis connection failed';
            } else if (memoryUsagePercent > 80) {
                status = HealthStatus.DEGRADED;
                message = `High Redis memory usage: ${memoryUsagePercent}%`;
            } else if (connectedClients > 100) {
                status = HealthStatus.DEGRADED;
                message = `High number of Redis clients: ${connectedClients}`;
            }

            return new HealthCheck('redis', status, message, details);
        } catch (error) {
            return new HealthCheck('redis', HealthStatus.UNHEALTHY, `Redis check failed: ${error.message}`);
        }
    }

    /**
     * Check package registry health
     */
    async checkPackageRegistry() {
        try {
            const registryConfig = this.config.registry;
            if (!registryConfig) {
                return new HealthCheck('package_registry', HealthStatus.UNHEALTHY, 'Registry configuration not found');
            }

            const healthUrl = `${registryConfig.url}/health`;
            const response = await this.makeHttpRequest(healthUrl);

            if (response.statusCode !== 200) {
                return new HealthCheck('package_registry', HealthStatus.UNHEALTHY, 
                    `Registry health endpoint returned status ${response.statusCode}`);
            }

            const healthData = JSON.parse(response.body);
            return new HealthCheck('package_registry', HealthStatus.HEALTHY, 'Package registry is healthy', healthData);
        } catch (error) {
            return new HealthCheck('package_registry', HealthStatus.UNHEALTHY, 
                `Package registry check failed: ${error.message}`);
        }
    }

    /**
     * Check CDN health
     */
    async checkCDN() {
        try {
            const cdnConfig = this.config.cdn;
            if (!cdnConfig) {
                return new HealthCheck('cdn', HealthStatus.UNHEALTHY, 'CDN configuration not found');
            }

            const healthUrl = `${cdnConfig.url}/health`;
            const response = await this.makeHttpRequest(healthUrl);

            if (response.statusCode !== 200) {
                return new HealthCheck('cdn', HealthStatus.UNHEALTHY, 
                    `CDN health check returned status ${response.statusCode}`);
            }

            const cdnHealth = JSON.parse(response.body);

            // Check sync status
            const syncUrl = `${cdnConfig.url}/sync/status`;
            const syncResponse = await this.makeHttpRequest(syncUrl);
            const syncStatus = JSON.parse(syncResponse.body);

            const details = {
                cdn_health: cdnHealth,
                sync_status: syncStatus
            };

            let status = HealthStatus.HEALTHY;
            let message = 'CDN is healthy';

            if (!syncStatus.synced) {
                status = HealthStatus.DEGRADED;
                message = 'CDN synchronization issues detected';
            }

            return new HealthCheck('cdn', status, message, details);
        } catch (error) {
            return new HealthCheck('cdn', HealthStatus.UNHEALTHY, `CDN check failed: ${error.message}`);
        }
    }

    /**
     * Check security scanning service
     */
    async checkSecurityScanning() {
        try {
            const securityConfig = this.config.security;
            if (!securityConfig) {
                return new HealthCheck('security_scanning', HealthStatus.UNHEALTHY, 'Security configuration not found');
            }

            const healthUrl = `${securityConfig.url}/health`;
            const response = await this.makeHttpRequest(healthUrl);

            if (response.statusCode !== 200) {
                return new HealthCheck('security_scanning', HealthStatus.UNHEALTHY, 
                    `Security service returned status ${response.statusCode}`);
            }

            const securityHealth = JSON.parse(response.body);

            let status = HealthStatus.HEALTHY;
            let message = 'Security scanning is healthy';

            if (!securityHealth.scanner_active) {
                status = HealthStatus.DEGRADED;
                message = 'Security scanner is not active';
            }

            return new HealthCheck('security_scanning', status, message, securityHealth);
        } catch (error) {
            return new HealthCheck('security_scanning', HealthStatus.UNHEALTHY, 
                `Security scanning check failed: ${error.message}`);
        }
    }

    /**
     * Make HTTP request
     */
    async makeHttpRequest(url) {
        return new Promise((resolve, reject) => {
            const client = url.startsWith('https') ? https : http;
            const req = client.get(url, { timeout: 10000 }, (res) => {
                let body = '';
                res.on('data', (chunk) => {
                    body += chunk;
                });
                res.on('end', () => {
                    resolve({
                        statusCode: res.statusCode,
                        body: body
                    });
                });
            });

            req.on('error', (error) => {
                reject(error);
            });

            req.on('timeout', () => {
                req.destroy();
                reject(new Error('Request timeout'));
            });
        });
    }

    /**
     * Run all health checks
     */
    async runAllChecks() {
        const report = new HealthReport();

        const checks = [
            await this.checkSystemResources(),
            await this.checkDatabase(),
            await this.checkRedis(),
            await this.checkPackageRegistry(),
            await this.checkCDN(),
            await this.checkSecurityScanning()
        ];

        checks.forEach(check => report.addCheck(check));

        return report;
    }
}

/**
 * Main function
 */
async function main() {
    // Configuration
    const config = {
        database: {
            host: process.env.DB_HOST || 'localhost',
            port: process.env.DB_PORT || '5432',
            name: process.env.DB_NAME || 'tusklang',
            user: process.env.DB_USER || 'postgres',
            password: process.env.DB_PASSWORD || ''
        },
        redis: {
            host: process.env.REDIS_HOST || 'localhost',
            port: process.env.REDIS_PORT || '6379',
            password: process.env.REDIS_PASSWORD || ''
        },
        registry: {
            url: process.env.REGISTRY_URL || 'http://localhost:8000'
        },
        cdn: {
            url: process.env.CDN_URL || 'https://cdn.tusklang.org'
        },
        security: {
            url: process.env.SECURITY_URL || 'http://localhost:9000'
        }
    };

    try {
        const checker = new TuskLangHealthChecker(config);
        const report = await checker.runAllChecks();

        // Output report as JSON
        console.log(JSON.stringify(report, null, 2));

        // Exit with appropriate code
        switch (report.overall_status) {
            case HealthStatus.UNHEALTHY:
                process.exit(1);
            case HealthStatus.DEGRADED:
                process.exit(2);
            default:
                process.exit(0);
        }
    } catch (error) {
        console.error('Health check failed:', error.message);
        process.exit(1);
    }
}

// Run if called directly
if (require.main === module) {
    main();
}

module.exports = {
    TuskLangHealthChecker,
    HealthCheck,
    HealthReport,
    HealthStatus
}; 

File: enterprise/health-checks/health-check.go
package main

import (
	"context"
	"database/sql"
	"encoding/json"
	"fmt"
	"log"
	"net/http"
	"os"
	"time"

	_ "github.com/lib/pq"
	"github.com/go-redis/redis/v8"
)

// HealthStatus represents the health status of a service
type HealthStatus string

const (
	StatusHealthy   HealthStatus = "healthy"
	StatusDegraded  HealthStatus = "degraded"
	StatusUnhealthy HealthStatus = "unhealthy"
	StatusUnknown   HealthStatus = "unknown"
)

// HealthCheck represents a single health check result
type HealthCheck struct {
	Name      string                 `json:"name"`
	Status    HealthStatus           `json:"status"`
	Message   string                 `json:"message"`
	Details   map[string]interface{} `json:"details,omitempty"`
	Timestamp time.Time              `json:"timestamp"`
}

// HealthReport represents the overall health report
type HealthReport struct {
	Timestamp     time.Time     `json:"timestamp"`
	OverallStatus HealthStatus  `json:"overall_status"`
	Checks        []HealthCheck `json:"checks"`
	Summary       struct {
		Total     int `json:"total"`
		Healthy   int `json:"healthy"`
		Degraded  int `json:"degraded"`
		Unhealthy int `json:"unhealthy"`
		Unknown   int `json:"unknown"`
	} `json:"summary"`
}

// HealthChecker manages health checks
type HealthChecker struct {
	config map[string]interface{}
}

// NewHealthChecker creates a new health checker
func NewHealthChecker(config map[string]interface{}) *HealthChecker {
	return &HealthChecker{
		config: config,
	}
}

// CheckSystemResources checks system resource usage
func (hc *HealthChecker) CheckSystemResources() HealthCheck {
	check := HealthCheck{
		Name:      "system_resources",
		Timestamp: time.Now(),
	}

	// In a real implementation, you would use system calls to get actual metrics
	// For now, we'll simulate the check
	cpuUsage := 45.0 // Simulated CPU usage
	memoryUsage := 60.0 // Simulated memory usage
	diskUsage := 75.0 // Simulated disk usage

	details := map[string]interface{}{
		"cpu_percent":    cpuUsage,
		"memory_percent": memoryUsage,
		"disk_percent":   diskUsage,
	}

	check.Details = details

	if cpuUsage > 80 || memoryUsage > 85 || diskUsage > 90 {
		check.Status = StatusDegraded
		check.Message = fmt.Sprintf("High resource usage - CPU: %.1f%%, Memory: %.1f%%, Disk: %.1f%%", cpuUsage, memoryUsage, diskUsage)
	} else {
		check.Status = StatusHealthy
		check.Message = "System resources are normal"
	}

	return check
}

// CheckDatabase checks database connectivity and performance
func (hc *HealthChecker) CheckDatabase() HealthCheck {
	check := HealthCheck{
		Name:      "database",
		Timestamp: time.Now(),
	}

	dbConfig, ok := hc.config["database"].(map[string]interface{})
	if !ok {
		check.Status = StatusUnhealthy
		check.Message = "Database configuration not found"
		return check
	}

	dsn := fmt.Sprintf("host=%s port=%s user=%s password=%s dbname=%s sslmode=disable",
		dbConfig["host"], dbConfig["port"], dbConfig["user"], dbConfig["password"], dbConfig["name"])

	db, err := sql.Open("postgres", dsn)
	if err != nil {
		check.Status = StatusUnhealthy
		check.Message = fmt.Sprintf("Database connection failed: %v", err)
		return check
	}
	defer db.Close()

	// Set connection timeout
	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
	defer cancel()

	// Test basic connectivity
	var result int
	err = db.QueryRowContext(ctx, "SELECT 1").Scan(&result)
	if err != nil {
		check.Status = StatusUnhealthy
		check.Message = fmt.Sprintf("Database query failed: %v", err)
		return check
	}

	// Check active connections
	var activeConnections int
	err = db.QueryRowContext(ctx, "SELECT count(*) FROM pg_stat_activity").Scan(&activeConnections)
	if err != nil {
		check.Status = StatusDegraded
		check.Message = fmt.Sprintf("Failed to get connection count: %v", err)
		return check
	}

	// Check database size
	var dbSize int64
	err = db.QueryRowContext(ctx, "SELECT pg_database_size(current_database())").Scan(&dbSize)
	if err != nil {
		check.Status = StatusDegraded
		check.Message = fmt.Sprintf("Failed to get database size: %v", err)
		return check
	}

	details := map[string]interface{}{
		"active_connections": activeConnections,
		"database_size_bytes": dbSize,
		"database_size_mb":   dbSize / 1024 / 1024,
	}

	check.Details = details

	if activeConnections > 100 {
		check.Status = StatusDegraded
		check.Message = fmt.Sprintf("High number of active connections: %d", activeConnections)
	} else {
		check.Status = StatusHealthy
		check.Message = "Database is healthy"
	}

	return check
}

// CheckRedis checks Redis connectivity and performance
func (hc *HealthChecker) CheckRedis() HealthCheck {
	check := HealthCheck{
		Name:      "redis",
		Timestamp: time.Now(),
	}

	redisConfig, ok := hc.config["redis"].(map[string]interface{})
	if !ok {
		check.Status = StatusUnhealthy
		check.Message = "Redis configuration not found"
		return check
	}

	rdb := redis.NewClient(&redis.Options{
		Addr:     fmt.Sprintf("%s:%s", redisConfig["host"], redisConfig["port"]),
		Password: redisConfig["password"].(string),
		DB:       0,
	})
	defer rdb.Close()

	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
	defer cancel()

	// Test basic connectivity
	_, err := rdb.Ping(ctx).Result()
	if err != nil {
		check.Status = StatusUnhealthy
		check.Message = fmt.Sprintf("Redis connection failed: %v", err)
		return check
	}

	// Get Redis info
	info, err := rdb.Info(ctx).Result()
	if err != nil {
		check.Status = StatusDegraded
		check.Message = fmt.Sprintf("Failed to get Redis info: %v", err)
		return check
	}

	// Parse info for key metrics
	var usedMemory, maxMemory, connectedClients int64
	fmt.Sscanf(info, "used_memory:%d\nmaxmemory:%d\nconnected_clients:%d", &usedMemory, &maxMemory, &connectedClients)

	memoryUsagePercent := float64(0)
	if maxMemory > 0 {
		memoryUsagePercent = float64(usedMemory) / float64(maxMemory) * 100
	}

	details := map[string]interface{}{
		"memory_usage_percent": memoryUsagePercent,
		"connected_clients":    connectedClients,
		"used_memory_mb":       usedMemory / 1024 / 1024,
		"max_memory_mb":        maxMemory / 1024 / 1024,
	}

	check.Details = details

	if memoryUsagePercent > 80 {
		check.Status = StatusDegraded
		check.Message = fmt.Sprintf("High Redis memory usage: %.1f%%", memoryUsagePercent)
	} else if connectedClients > 100 {
		check.Status = StatusDegraded
		check.Message = fmt.Sprintf("High number of Redis clients: %d", connectedClients)
	} else {
		check.Status = StatusHealthy
		check.Message = "Redis is healthy"
	}

	return check
}

// CheckPackageRegistry checks package registry service health
func (hc *HealthChecker) CheckPackageRegistry() HealthCheck {
	check := HealthCheck{
		Name:      "package_registry",
		Timestamp: time.Now(),
	}

	registryConfig, ok := hc.config["registry"].(map[string]interface{})
	if !ok {
		check.Status = StatusUnhealthy
		check.Message = "Registry configuration not found"
		return check
	}

	registryURL := registryConfig["url"].(string)

	// Check health endpoint
	client := &http.Client{Timeout: 10 * time.Second}
	resp, err := client.Get(registryURL + "/health")
	if err != nil {
		check.Status = StatusUnhealthy
		check.Message = fmt.Sprintf("Registry health check failed: %v", err)
		return check
	}
	defer resp.Body.Close()

	if resp.StatusCode != http.StatusOK {
		check.Status = StatusUnhealthy
		check.Message = fmt.Sprintf("Registry health endpoint returned status %d", resp.StatusCode)
		return check
	}

	// Parse health response
	var healthData map[string]interface{}
	if err := json.NewDecoder(resp.Body).Decode(&healthData); err != nil {
		check.Status = StatusDegraded
		check.Message = fmt.Sprintf("Failed to parse health response: %v", err)
		return check
	}

	check.Details = healthData
	check.Status = StatusHealthy
	check.Message = "Package registry is healthy"

	return check
}

// CheckCDN checks CDN health and synchronization
func (hc *HealthChecker) CheckCDN() HealthCheck {
	check := HealthCheck{
		Name:      "cdn",
		Timestamp: time.Now(),
	}

	cdnConfig, ok := hc.config["cdn"].(map[string]interface{})
	if !ok {
		check.Status = StatusUnhealthy
		check.Message = "CDN configuration not found"
		return check
	}

	cdnURL := cdnConfig["url"].(string)

	// Check CDN health
	client := &http.Client{Timeout: 10 * time.Second}
	resp, err := client.Get(cdnURL + "/health")
	if err != nil {
		check.Status = StatusUnhealthy
		check.Message = fmt.Sprintf("CDN health check failed: %v", err)
		return check
	}
	defer resp.Body.Close()

	if resp.StatusCode != http.StatusOK {
		check.Status = StatusUnhealthy
		check.Message = fmt.Sprintf("CDN health check returned status %d", resp.StatusCode)
		return check
	}

	// Parse CDN health response
	var cdnHealth map[string]interface{}
	if err := json.NewDecoder(resp.Body).Decode(&cdnHealth); err != nil {
		check.Status = StatusDegraded
		check.Message = fmt.Sprintf("Failed to parse CDN health response: %v", err)
		return check
	}

	// Check sync status
	resp, err = client.Get(cdnURL + "/sync/status")
	if err != nil {
		check.Status = StatusDegraded
		check.Message = fmt.Sprintf("CDN sync status check failed: %v", err)
		return check
	}
	defer resp.Body.Close()

	var syncStatus map[string]interface{}
	if err := json.NewDecoder(resp.Body).Decode(&syncStatus); err != nil {
		check.Status = StatusDegraded
		check.Message = fmt.Sprintf("Failed to parse sync status: %v", err)
		return check
	}

	details := map[string]interface{}{
		"cdn_health":  cdnHealth,
		"sync_status": syncStatus,
	}

	check.Details = details

	if synced, ok := syncStatus["synced"].(bool); !ok || !synced {
		check.Status = StatusDegraded
		check.Message = "CDN synchronization issues detected"
	} else {
		check.Status = StatusHealthy
		check.Message = "CDN is healthy"
	}

	return check
}

// CheckSecurityScanning checks security scanning service health
func (hc *HealthChecker) CheckSecurityScanning() HealthCheck {
	check := HealthCheck{
		Name:      "security_scanning",
		Timestamp: time.Now(),
	}

	securityConfig, ok := hc.config["security"].(map[string]interface{})
	if !ok {
		check.Status = StatusUnhealthy
		check.Message = "Security configuration not found"
		return check
	}

	securityURL := securityConfig["url"].(string)

	// Check security service health
	client := &http.Client{Timeout: 10 * time.Second}
	resp, err := client.Get(securityURL + "/health")
	if err != nil {
		check.Status = StatusUnhealthy
		check.Message = fmt.Sprintf("Security service health check failed: %v", err)
		return check
	}
	defer resp.Body.Close()

	if resp.StatusCode != http.StatusOK {
		check.Status = StatusUnhealthy
		check.Message = fmt.Sprintf("Security service returned status %d", resp.StatusCode)
		return check
	}

	// Parse security health response
	var securityHealth map[string]interface{}
	if err := json.NewDecoder(resp.Body).Decode(&securityHealth); err != nil {
		check.Status = StatusDegraded
		check.Message = fmt.Sprintf("Failed to parse security health response: %v", err)
		return check
	}

	check.Details = securityHealth

	if scannerActive, ok := securityHealth["scanner_active"].(bool); !ok || !scannerActive {
		check.Status = StatusDegraded
		check.Message = "Security scanner is not active"
	} else {
		check.Status = StatusHealthy
		check.Message = "Security scanning is healthy"
	}

	return check
}

// RunAllChecks runs all health checks and returns a comprehensive report
func (hc *HealthChecker) RunAllChecks() HealthReport {
	checks := []HealthCheck{
		hc.CheckSystemResources(),
		hc.CheckDatabase(),
		hc.CheckRedis(),
		hc.CheckPackageRegistry(),
		hc.CheckCDN(),
		hc.CheckSecurityScanning(),
	}

	// Calculate overall status
	overallStatus := StatusHealthy
	summary := struct {
		Total     int `json:"total"`
		Healthy   int `json:"healthy"`
		Degraded  int `json:"degraded"`
		Unhealthy int `json:"unhealthy"`
		Unknown   int `json:"unknown"`
	}{
		Total: len(checks),
	}

	for _, check := range checks {
		switch check.Status {
		case StatusHealthy:
			summary.Healthy++
		case StatusDegraded:
			summary.Degraded++
			if overallStatus == StatusHealthy {
				overallStatus = StatusDegraded
			}
		case StatusUnhealthy:
			summary.Unhealthy++
			overallStatus = StatusUnhealthy
		case StatusUnknown:
			summary.Unknown++
			overallStatus = StatusUnhealthy
		}
	}

	report := HealthReport{
		Timestamp:     time.Now(),
		OverallStatus: overallStatus,
		Checks:        checks,
		Summary:       summary,
	}

	return report
}

func main() {
	// Configuration (in production, load from environment or config file)
	config := map[string]interface{}{
		"database": map[string]interface{}{
			"host":     "localhost",
			"port":     "5432",
			"name":     "tusklang",
			"user":     "postgres",
			"password": "",
		},
		"redis": map[string]interface{}{
			"host":     "localhost",
			"port":     "6379",
			"password": "",
		},
		"registry": map[string]interface{}{
			"url": "http://localhost:8000",
		},
		"cdn": map[string]interface{}{
			"url": "https://cdn.tusklang.org",
		},
		"security": map[string]interface{}{
			"url": "http://localhost:9000",
		},
	}

	checker := NewHealthChecker(config)
	report := checker.RunAllChecks()

	// Output report as JSON
	jsonData, err := json.MarshalIndent(report, "", "  ")
	if err != nil {
		log.Fatal("Failed to marshal report:", err)
	}

	fmt.Println(string(jsonData))

	// Exit with appropriate code
	switch report.OverallStatus {
	case StatusUnhealthy:
		os.Exit(1)
	case StatusDegraded:
		os.Exit(2)
	default:
		os.Exit(0)
	}
} 